src/onegov/org/forms/newsletter.py

```diff
@@ -212,3 +220,102 @@ def recipient(self):
                     .one()
 
         return NewsletterSendFormWithRecipients
+
+
+class NewsletterSubscriberImportForm(Form):
+
+    dry_run = BooleanField(
+        label=_("Dry Run"),
+        description=_("Do not actually import the newsletter subscribers"),
+        default=False
+    )
+
+    file = UploadField(
+        label=_("Import"),
+        validators=[
+            DataRequired(),
+            WhitelistedMimeType({
+                'application/excel',
+                'application/vnd.ms-excel',
+                (
+                    'application/'
+                    'vnd.openxmlformats-officedocument.spreadsheetml.sheet'
+                ),
+                'application/vnd.ms-office',
+                'application/octet-stream',
+                'application/zip',
+                'text/csv',
+                'text/plain',
+            }),
+            FileSizeLimit(10 * 1024 * 1024)
+        ],
+        render_kw=dict(force_simple=True)
+    )
+
+    @property
+    def headers(self):
+        return {
+            'address': self.request.translate(_("Address")),
+        }
+
+    def run_export(self):
+        recipients = RecipientCollection(self.request.session)
+        headers = self.headers
+
+        def get(recipient, attribute):
+            result = (getattr(recipient, attribute, None))
+            result = result.strip()
+            return result
+
+        result = []
+        for recipient in recipients.query():
+            result.append({
+                v: get(recipient, k)
+                for k, v in headers.items()
+            })
+        return result
+
+    def run_import(self):
+        headers = self.headers
+        session = self.request.session
+        recipients = RecipientCollection(session)
+        csvfile = convert_excel_to_csv(self.file.file)
```

Comment:
As far as I remember this function might throw a few exceptions which you might want to catch.... (https://github.com/OneGov/onegov-cloud/blob/7c06b111f30e73e8ed075bcb151ca7cb8b86b584/src/onegov/election_day/formats/common.py#L61)